data_structures/src/chain.rs

```diff
@@ -199,7 +205,7 @@ impl<T: AsRef<[u8]>> Hashable for T {
 
 impl Hashable for Block {
     fn hash(&self) -> Hash {
-        calculate_sha256(&self.block_header.to_pb_bytes().unwrap()).into()
+        calculate_sha256(&self.proof.to_pb_bytes().unwrap()).into()
```

Comment:
Please revert this change. If you want unique block hashes for testing, change the mint pkh in mining.rs:370 to a random value:

```rust
let pkh = thread_rng().gen();
```